access/pagerduty/app.go

```diff
@@ -238,8 +238,9 @@ func (a *App) onPendingRequest(ctx context.Context, req types.AccessRequest) err
 	}
 
 	var (
-		resultErr error
-		data      PagerdutyData
+		notifyErr  error
```

Comment:
I found this method to be very confusing with all the nesting and variables floating around. What do you think about the following? It's functionally equivalent with just some changes to the logs/comments.
```
func (a *App) onPendingRequest(ctx context.Context, req types.AccessRequest) error {
	if len(req.GetSystemAnnotations()) == 0 {
		logger.Get(ctx).Debug("Cannot proceed further. Request is missing any annotations")
		return nil
	}

	// First, try to create a notification incident.
	var notifyErr error
	serviceName, err := a.getNotifyServiceName(req)
	if err != nil {
		logger.Get(ctx).Debugf("Could not determine notification service info, skipping notification: %s", err)
		return trace.Wrap(a.tryApproveRequest(ctx, req, PagerdutyData{}))
	}

	// To minimize the count of auto-approval tries, attempt approval only when
	// the incident is new. If there's an error, attempt approval anyways since
	// we can't tell if it's a new incident or not.
	data, isNew, notifyErr := a.tryNotifyService(ctx, req, serviceName)
	if notifyErr == nil && !isNew {
		return nil
	}

	approveErr := trace.Wrap(a.tryApproveRequest(ctx, req, data))
	return trace.NewAggregate(notifyErr, approveErr)
}

func (a *App) tryApproveRequest(ctx context.Context, req types.AccessRequest, data PagerdutyData) error {
	log := logger.Get(ctx)
	serviceNames, err := a.getOnCallServiceNames(req)
	if err != nil {
		logger.Get(ctx).Debugf("Could not determine approval service info, skipping approval: %s", err)
		return nil
	}
	...

Reply:
Moved annotation digging back to `tryNotifyService` and `tryApproveRequest`.